rating/peersRatingMonitor.go

```diff
@@ -0,0 +1,89 @@
+package rating
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	p2p "github.com/multiversx/mx-chain-p2p-go"
+	"github.com/multiversx/mx-chain-storage-go/types"
+)
+
+// ArgPeersRatingMonitor is the DTO used to create a new peers rating monitor
+type ArgPeersRatingMonitor struct {
+	TopRatedCache types.Cacher
+	BadRatedCache types.Cacher
+}
+
+type peersRatingMonitor struct {
+	topRatedCache types.Cacher
+	badRatedCache types.Cacher
+}
+
+// NewPeersRatingMonitor returns a new peers rating monitor
+func NewPeersRatingMonitor(args ArgPeersRatingMonitor) (*peersRatingMonitor, error) {
+	err := checkMonitorArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	return &peersRatingMonitor{
+		topRatedCache: args.TopRatedCache,
+		badRatedCache: args.BadRatedCache,
+	}, nil
+}
+
+func checkMonitorArgs(args ArgPeersRatingMonitor) error {
+	if check.IfNil(args.TopRatedCache) {
+		return fmt.Errorf("%w for TopRatedCache", p2p.ErrNilCacher)
+	}
+	if check.IfNil(args.BadRatedCache) {
+		return fmt.Errorf("%w for BadRatedCache", p2p.ErrNilCacher)
+	}
+
+	return nil
+}
+
+// GetPeersRatings returns the current ratings
+func (monitor *peersRatingMonitor) GetPeersRatings() string {
+	ratingsMap := getRatings(monitor.topRatedCache)
```

Comment:
we need to filter out the data. We should first get the connected peers list and then extract the bad/good rating of those connected peers. To achieve this, we need a new small interface that defines the Connected() []core.PeerID function (which is implemented by the netMessenger)
Otherwise, without this filtering, the response is too large (possibly containing ~10000 elements)
This function then can be renamed to GetConnectedPeersRatings

Reply:
pushed